### PR TITLE
Adding scroll percentage and time on page analytics

### PR DESCRIPTION
**Pull request by robert-j-webb**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -308,5 +308,45 @@ $(document).ready(function () {
...
+  setInterval(() => {
+    const curScroll = scrollPercentage();
+    if (curScroll > maxScroll) {
+      maxScroll = curScroll;
+    }
+  }, 500);
+
+  window.addEventListener('beforeunload', () => {
+    trackTimeOnPage(window.location.pathname);
+    amplitude.track('MaxScrollPercentage', { maxScroll });
```
Comment (serenackuo) [inline src/index.js:347]: Should we pass the page location into the properties?

File: src/index.js [hunk 2]
```diff
@@ -308,5 +308,45 @@ $(document).ready(function () {
...
+  }
+
+  function scrollPercentage() {
+    const { documentElement } = document,
+      { body } = document;
+
+    return (
+      ((documentElement.scrollTop || body.scrollTop) /
+        ((documentElement.scrollHeight || body.scrollHeight) - documentElement.clientHeight)) *
+      100
```
Comment (serenackuo) [inline src/index.js:332]: Should we Math.floor this value to make it a little more readable in Amplitude?

**Review by serenackuo — COMMENTED** [review]
We should probably add new events into Amplitude, along with their properties, so that they're not listed as unexpected.

**Comment by robert-j-webb** [comment]
> We should probably add new events into Amplitude, along with their properties, so that they're not listed as unexpected.

I do this weekly, when I get the health report from amplitude

**Pull request merged**